odl/tomo/geometry/conebeam.py

```diff
@@ -16,13 +16,457 @@
 
 import numpy as np
 
-from odl.tomo.geometry.detector import Flat2dDetector
+from odl.tomo.geometry.detector import Flat1dDetector, Flat2dDetector
 from odl.tomo.geometry.geometry import (
     DivergentBeamGeometry, AxisOrientedGeometry)
-from odl.tomo.util.utility import perpendicular_vector
+from odl.tomo.util.utility import euler_matrix, transform_system
+from odl.util import signature_string, indent_rows
 
 
-__all__ = ('CircularConeFlatGeometry', 'HelicalConeFlatGeometry',)
+__all__ = ('FanFlatGeometry',
+           'CircularConeFlatGeometry', 'HelicalConeFlatGeometry',)
+
+
+class FanFlatGeometry(DivergentBeamGeometry):
+
+    """Fan beam (2d cone beam) geometry with flat 1d detector.
+
+    The source moves on a circle with radius ``src_radius``, and the
+    detector reference point is opposite to the source, i.e. at maximum
+    distance, on a circle with radius ``det_radius``. One of the two
+    radii can be chosen as 0, which corresponds to a stationary source
+    or detector, respectively.
+
+    The motion parameter is the 1d rotation angle parameterizing source
+    and detector positions simultaneously.
+
+    In the standard configuration, the detector is perpendicular to the
+    ray direction, its reference point is initially at ``(0, 1)``, and
+    the initial detector axis is ``(1, 0)``.
+
+    For details, check `the online docs
+    <https://odlgroup.github.io/odl/guide/geometry_guide.html>`_.
+    """
+
+    _default_config = dict(src_to_det_init=(0, 1), det_axis_init=(1, 0))
+
+    def __init__(self, apart, dpart, src_radius, det_radius,
+                 src_to_det_init=(0, 1), **kwargs):
+        """Initialize a new instance.
+
+        Parameters
+        ----------
+        apart : 1-dim. `RectPartition`
+            Partition of the angle interval.
+        dpart : 1-dim. `RectPartition`
+            Partition of the detector parameter interval.
+        src_radius : nonnegative float
+            Radius of the source circle.
+        det_radius : nonnegative float
+            Radius of the detector circle.
+        src_to_det_init : `array-like` (shape ``(2,)``), optional
+            Initial state of the vector pointing from source to detector
+            reference point. The zero vector is not allowed.
+        det_axis_init : `array-like` (shape ``(2,)``), optional
```

Comment:
Even if there's only one? Fine with me, but then we should do it in other places, too.